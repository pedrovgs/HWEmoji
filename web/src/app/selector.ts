export function $$(selector: string, context: Document) {
  context = context || document;
  var elements = context.querySelectorAll(selector);
  return Array.prototype.slice.call(elements);
}
