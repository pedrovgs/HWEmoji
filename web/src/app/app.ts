import { $$ } from "./selector";

interface Props {
  className: string;
  title: string;
  content: string;
}
export function init(props: Props) {
  const { className, title, content } = props;
  const main = document.querySelector(className);
  const height = window.innerHeight * 0.65;
  const width = height * 0.75;
  const canvas = document.createElement("canvas") as HTMLCanvasElement;
  Object.assign(canvas, { width, height });
  if (main) {
    main.appendChild(canvas);
  }
  const img = new Image();
  img.src = content;
  const context = canvas.getContext("2d");

  if (context) {
    img.onload = function () {
      context.drawImage(img, 0, 0, width, height);
      console.log("drawn");
    };
  }
}
