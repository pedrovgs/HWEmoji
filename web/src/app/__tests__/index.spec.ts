import { TextEncoder, TextDecoder } from "util";
global.TextEncoder = TextEncoder;
// @ts-ignore
global.TextDecoder = TextDecoder;

import { JSDOM } from "jsdom";
import { init } from "../app";

const { window } = new JSDOM("<!doctype html><html><body></body></html>");
// Save these two objects in the global space so that libraries/tests
// can hook into them, using the above doc definition.
global.document = window.document;
// global.window = window as typeof globalThis;
global.Image = window.Image;

describe("app", () => {
  beforeEach(() => {
    window.document.body.innerHTML = `
    <header class="heading"></header>
    <section class="section-1"></section>
    <script src="./bundle.js"></script>`;
  });
  it("should render without error", () => {
    init({
      className: ".card",
      title: "Beckhi",
      content: "./public/images/beckhi.jpg",
    });
  });
  it("should match snapshot", () => {
    init({
      className: ".card",
      title: "Beckhi",
      content: "./public/images/beckhi.jpg",
    });
    expect(window.document.body).toMatchSnapshot();
  });
});
