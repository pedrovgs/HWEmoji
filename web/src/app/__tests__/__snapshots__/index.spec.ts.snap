// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should match snapshot 1`] = `
<body>
  
    
  <header
    class="heading"
  />
  
    
  <section
    class="section-1"
  />
  
    
  <script
    src="./bundle.js"
  />
</body>
`;
